python/lsst/pipe/analysis/analysis.py

```diff
@@ -1094,6 +1098,91 @@ def plotQuiver(self, catalog, description, plotInfoDict, areaDict, log, cmap=plt
 
         yield Struct(fig=fig, description=description, stats=stats, statsHigh=None, dpi=150, style="quiver")
 
+    def plotRhoStatistics(self, description, plotInfoDict, log, treecorrParams,
+                          stats, butler=None, camera=None,
+                          ccdList=None, tractInfo=None, patchList=None, hscRun=None, matchRadius=None,
+                          zpLabel=None, forcedStr=None, postFix="", flagsCat=None, uberCalLabel=None):
+        """ Plot Rho Statistics """
+
+        figAxes = [plt.subplots(), plt.subplots()]  # first plot for Rho 1, 3, 4 and second for Rho 2, 5
+        figs, axes = list(zip(*figAxes))
+
+        compareCol = "base_SdssShape"
+        psfCompareCol = "base_SdssShape_psf"
+        shapeAlgorithm = "SDSS"
+        flags = list(self.config.flags) + ["base_SdssShape_flag", "base_SdssShape_flag_psf"]
+
+        # Cull the catalog of flagged sources
+        bad = np.zeros(len(self.catalog), dtype=bool)
+        bad |= self.catalog["deblend_nChild"] > 0
+        for flag in flags:
+            bad |= self.catalog[flag]
+        good_catalog = self.catalog[~bad].copy(deep=True)
+
+        rhoStatsFunc = RhoStatistics(compareCol, psfCompareCol, **treecorrParams)
+        rhoStats = rhoStatsFunc(good_catalog)
+        plotRhoStats(axes, rhoStats)
+
+        for figId, figax in enumerate(figAxes):
+            fig, ax = figax
+            figDescription = description + str(figId + 1)
+            labelCamera(plotInfoDict, fig, ax, 0.5, 1.09)
+            labelVisit(plotInfoDict, fig, ax, 0.5, 1.04)
+            if zpLabel is not None:
+                plotText(zpLabel, fig, ax, 0.14, -0.08, prefix="zp: ", color="green")
+            if uberCalLabel:
+                plotText(uberCalLabel, fig, ax, 0.14, -0.12, prefix="uberCal: ", fontSize=7, color="green")
+            plotText(shapeAlgorithm, fig, ax, 0.85, -0.08, prefix="Shape Alg: ", fontSize=8, color="green")
+            if forcedStr is not None:
+                plotText(forcedStr, fig, ax, 0.85, -0.12, prefix="cat: ", fontSize=8, color="green")
+
+            log.info("Tract id in Rho Stats: {0}".format(plotInfoDict['tract']))
+
+            yield Struct(fig=fig, description=figDescription, stats=stats, statsHigh=None, dpi=500,
+                         style="RhoStats")
+
+        if "ext_shapeHSM_HsmSourceMoments_xx" in self.catalog.schema:
```

Comment:
This seems like quite a bit of code duplication, is it possible to refactor to avoid it? Maybe make a new function that takes, psfCompareCol, compareCol, shapeAlgorithm and flags? Or cycle over them?

Reply:
Going by Jim's thumb rule, it is okay to have code duplication twice. So I'll leave it as it is.